MP1 update (#774)
stm32duino/Arduino_Core_STM32: variants/STM32MP157_DK/README.md

* [STM32 MPU OpenSTLinux Distribution]
* [Balena OS]

> __Note that the first two OSes should select `stm32mp157c-dk2-m4-examples-sdcard` boot mode (or device tree configuration) on boot time.__ `stm32mp157c-dk2-sdcard` boot mode is also known to work but it is not guaranteed. See [ST Wiki page on boot mode] for more detail. If you are using it for your own OS, see the [Linux Device Tree considerations](#Linux-Device-Tree-considerations) section.

Reviewer: Because our generate script is quite generic we can add this note?
```suggestion
Other distributions not listed here may run Arduino but they are not tested. __Note that the first two OSes should select `stm32mp157c-dk2-m4-examples-sdcard` boot mode (or device tree configuration) on boot time.__ `stm32mp157c-dk2-sdcard` boot mode is also known to work but it is not guaranteed. See [ST Wiki page on boot mode] for more detail. If you are using it for your own OS, see the [Linux Device Tree considerations](#Linux-Device-Tree-considerations) section.
```

Author: Note added

Reviewer: Great, thanks for updating! :)

Author: Welcome 😉 
Thanks too for your review. 👍

## How to use

After Verify and Upload, you will see a message similar to the following in Arduino IDE:

    <Arduino build output path>/run_arduino_<sketch name>.sh generated successfully.
    This file should be uploaded manually by SCP, SFTP, Kermit, or etc.
    Then run "sh ./run_arduino_<sketch name>.sh start" command in the board's console.
    For detailed instructions, please visit:
        https://github.com/stm32duino/Arduino_Core_STM32/tree/master/variants/STM32MP157_DK/README.md

> In this example, the user **must** upload `<Arduino build output path>/run_arduino_<sketch name>.sh` file manually. Uploading instruction is described later in the [Uploading](#Uploading) section. 

Reviewer: In my opinion, the new name of `<Arduino build output path>/run_arduino_<your project name>.sh` reduces readability of the document and users may take more time to notice what `<Arduino build output path>/run_arduino_<your project name>.sh` is. So I think adding example samples of `<Arduino build output path>/run_arduino_<your project name>.sh` will help the users. What about adding the example after this paragraph?
```markdown
`<Arduino build output path>/run_arduino_<your project name>.sh` looks like this:

* `C:/Users/kbumsik/AppData/Local/Temp/arduino_build_668148/run_arduino.Blink.ino.sh` (For Windows)
* `/tmp/arduino_build_668148/run_arduino_Blink.ino.sh` (For Linux/macOS)
```

* [STM32 MPU OpenSTLinux Distribution]
* [Balena OS]

#### Note
 * Other distributions not listed here may run Arduino firmware but they are not tested.
 * __The first two OSes should select `stm32mp157c-dk2-m4-examples-sdcard` boot mode (or device tree configuration) on boot time.__
 * `stm32mp157c-dk2-sdcard` boot mode is also known to work but it is not guaranteed. See [ST Wiki page on boot mode] for more detail. If you are using it for your own OS, see the [Linux Device Tree considerations](#Linux-Device-Tree-considerations) section.

## How to use

After Verify and Upload, you will see a message similar to the following in Arduino IDE:

    <Arduino build output path>/run_arduino_<sketch name>.sh generated successfully.
    This file should be uploaded manually by SCP, SFTP, Kermit, or etc.
    Then run "sh ./run_arduino_<sketch name>.sh start" command in the board's console.
    For detailed instructions, please visit:
        https://github.com/stm32duino/Arduino_Core_STM32/tree/master/variants/STM32MP157_DK/README.md

> `<Arduino build output path>/run_arduino_<your project name>.sh` looks like this for *Blink* example:

Reviewer: ```suggestion
`<Arduino build output path>/run_arduino_<sketch name>.sh` looks like this for *Blink* example:
```

Author: fixed

#### Note
 * `sh run_arduino_<sketch name>.sh start` is a one-shot command, the Arduino firmware only runs for the current boot. If you want to make it run after reboot, you need to use `sh run_arduino_<sketch name>.sh install` command. 

`run_arduino_<sketch name>.sh` help page summary:

    Usage: sh run_arduino_<sketch name>.sh [start|stop|restart|generate|install|uninstall]

    run_arduino_<sketch name>.sh is a helper script that helps managing an Arduino binary
    file for the coprocessor using remoteproc framework.

    sh run_arduino_<sketch name>.sh start
        Upload the binary to the coprocessor then start it.
        This command must be executed while the script contains the binary
        after generate command is run.

    sh run_arduino_<sketch name>.sh install
        Run the binary on boot automatically by installing a systemd service.

    sh run_arduino_<sketch name>.sh uninstall
        Uninstall the autostart service.

    sh run_arduino_<sketch name>.sh stop
        Stop the coprocessor.

    sh run_arduino_<sketch name>.sh restart
        Restart the coprocessor.

> See the source code for the full help page and the more details about [run_arduino_<sketch name>.sh].

Reviewer: It looks like `<>` in link does not work, rendering like `run_arduino_`.
```suggestion
See the source code for the full help page and the more details about [run_arduino_gen.sh].
```

Author: I've updated and add a sentence about the gen script to be more precise.

## Uploading

> As mentioned above `run_arduino.sh` file have to be uploaded manually to the Linux file system in order to be executed on the board directly through the Linux console. There are many ways to upload the file.

Reviewer: What about using easier words like "From the PC to the board"?
```suggestion
As mentioned above `run_arduino.sh` file have to be uploaded manually from the PC to the board's Linux file system in order to be executed on the board directly through the board's Linux console. There are many ways to upload the file.
```

[Balena OS]: https://github.com/kbumsik/balena-st-stm32mp
[ST Wiki page on boot mode]: https://wiki.st.com/stm32mpu/wiki/STM32CubeMP1_Package#Getting_started_with_STM32CubeMP1_Package

> [run_arduino_<sketch name>.sh]: https://github.com/stm32duino/Arduino_Tools/blob/master/linux/run_arduino_gen.sh

Reviewer: ```suggestion
[run_arduino_gen.sh]: https://github.com/stm32duino/Arduino_Tools/blob/master/run_arduino_gen.sh
```

Author: ```
See the source code run_arduino_gen.sh for the full help page and the more details about the run_arduino_<sketch name>.sh generated.

run_arduino_gen.sh is the shell script that produces a copy of the script called run_arduino_<sketch name>.sh but with the sketch binary self-contained.
```
